experimental/packages/opentelemetry-sdk-metrics-base/src/MeterProvider.ts

```diff
@@ -48,16 +45,24 @@ export class MeterProvider {
         api.diag.warn('A shutdown MeterProvider cannot provide a Meter')
         return metrics.NOOP_METER;
     }
+    // TODO: add schemaUrl to id?
+    const id = `${name}#${version}`;
+    if (this._sharedState.meters.has(id)) {
+      return this._sharedState.meters.get(id)!;
+    }
 
     // Spec leaves it unspecified if creating a meter with duplicate
```

Comment:
This comment is no longer accurate

Reply:
Extracted this part of change to https://github.com/open-telemetry/opentelemetry-js/pull/2666 as it is not the essential part of the metric streams.